### DISCUSSION TITLE
Calculation of thermodynamic and physical properties in Java/Python/Matlab/C#

### DISCUSSION THREAD

**Discussion by EvenSol** [equinor/neqsim#199] [use of NeqSim]
Example of creating a fluid and calculating thermodynamic and physical properties in Java:
[Java example](https://github.com/equinor/neqsim/blob/master/src/test/java/neqsim/thermo/util/example/TestFlash.java)

Example of creating a fluid and calculating thermodynamic and physical properties in Python:
[Python example](https://github.com/equinor/neqsimpython/blob/master/examples/propertiesOfNaturalGas.py)

Example of creating a fluid and calculating thermodynamic and physical properties in Matlab:
[Matlab example](https://github.com/equinor/neqsimmatlab/blob/master/example/GasFlashProperties.m)

Example of creating a fluid and calculating thermodynamic and physical properties in C#:
[C# example](https://github.com/equinor/neqsimNET/blob/master/examples/TPflash.cs)

**Comment by UniqASL** [comment]
Is there a way to define a fluid with other properties than temperature and pressure?

Said differently, I calculate for a given point (p1, T1) the fluid entropy s1 and I would like to find for another pressure p2 the fluid temperature T2 associated to the entropy s1.

Is there a clever way to do it, other than just iterating on T2 until s2 = s1?

Thanks in advance for your assistance!

**Reply by EvenSol** [reply]
Yes - in this case you can use the constant entropy flash (PSflash).

Example in Java:
https://github.com/equinor/neqsim/blob/master/src/test/java/neqsim/thermo/util/example/PS_PH_flash.java

Example in Python:
https://colab.research.google.com/github/EvenSol/NeqSim-Colab/blob/master/notebooks/thermodynamics/RachRice.ipynb

Matlab:
https://github.com/equinor/neqsimmatlab/blob/master/example/TP_PH_flash.m

Various flash calculations in NeqSim are seen here:
https://github.com/equinor/neqsim/blob/master/src/main/java/neqsim/thermodynamicOperations/ThermodynamicOperations.java

**Reply by EvenSol** [reply]
```
Example of a Python script:

nitrogen = 1.0
CO2 = 2.5
methane = 80.0
ethane = 5.0
propane =  2.5
ibutane =  1.25 
nbutane =  1.0
ipentane =  0.4 
npentane =  0.3 
nhexane =  0.08

fluid1 = fluid('srk')
fluid1.addComponent("nitrogen", nitrogen)
fluid1.addComponent("CO2", CO2)
fluid1.addComponent("methane", methane)
fluid1.addComponent("ethane", ethane)
fluid1.addComponent("propane", propane)
fluid1.addComponent("i-butane", ibutane)
fluid1.addComponent("n-butane", nbutane)
fluid1.addComponent("i-pentane", ipentane)
fluid1.addComponent("n-pentane", nbutane)
fluid1.addComponent("n-hexane", nhexane)
fluid1.setMixingRule('classic')

fluid1.setTemperature(20.0, 'C')
fluid1.setPressure(50.0, 'bara')

TPflash(fluid1)
fluid1.initProperties()
Sunit = 'J/kgK'
entropy = fluid1.getEntropy(Sunit)

fluid1.setPressure(10.0, 'bara')

PSflash(fluid1, entropy,unit=Sunit)

endTemperature = fluid1.getTemperature('C')

```

**Reply by UniqASL** [reply]
Thanks a lot!
